### PR TITLE
Dump database to home directory

### PR DESCRIPTION
**Pull request by JamelleG**
Open this as a general PR in order to gather some feedback, it was mentioned that we had some issues with keeping databases in project directory (mainly with unison, which we could probably ignore, no?).
By default `wf2 db-dump` dumps to the current project path.
I have started some work on changing this to the home directory. 

Still need to work out whether we use the context we are currently in and format it as `$HOME/.wf2/db/{PROJECT}.sql` or any other suggestions on this would be appreciated.

## TODO: 
- Ensure path exist if not create it.
- Work out how we handle this if every project is dumping to the same path.

### CODE REVIEW

**Comment by shakyShane** [comment]
@JamelleG make the dump a combo of a date + name?

eg: `ukmeds-01-02-2019`

File: wf2_core/src/recipes/m2/mod.rs [hunk 1]
```diff
@@ -283,16 +283,23 @@ impl M2Recipe {
     pub fn db_dump(&self, ctx: &Context) -> Vec<Task> {
         use m2_env::{DB_NAME, DB_PASS, DB_USER};
         let container_name = format!("wf2__{}__db", ctx.name);
+
+        //TODO: think about how we want to handle the case when a home directory can,
```
Comment (shakyShane) [inline wf2_core/src/recipes/m2/mod.rs]: I don't think we need to worry about that. All users will be on MacOS, so just like with `current_dir()` I think there are cases where if it doesn't work we just bail

Comment (JamelleG) [reply wf2_core/src/recipes/m2/mod.rs]: Yes, I think in this case if there is no homedir we can fallback to the current dir, ensuring they can still continue as normal.
I use this on linux, although I don't really need the benefits that unison provides in terms of speed, it offers a standard environment and set of commands.

Comment (shakyShane) [reply wf2_core/src/recipes/m2/mod.rs]: doesn't `current_dir()` work on linux?

**Pull request closed without merging**